lib/dialects/mssql/query-generator.js

```diff
@@ -132,7 +132,7 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
           const commentMatch = dataType.match(/^(.+) (COMMENT.*)$/);
           const commentText = commentMatch[2].replace(/COMMENT/, '').trim();
           commentStr += _.template(commentTemplate, this._templateSettings)({
-            table: this.quoteIdentifier(tableName),
+            table: this.quoteIdentifier(options.temporaryTable ? `##${tableName}` : tableName),
```

Comment:
`##` is a global temporary table which is shared by all sessions and is available till last session using it. We should go for `#` local temporary table which acts like temporary table of other dialects i.e. limited to current session

Reply:
Thanks, looks like I need to dig a little more to come up with the best way to handle MSSQL. Two tests failed and I am having a hard time getting the docker image for it to run.

Reply:
Docker image will for anything transaction (unfortunately I don't have any real experience with MSSQL either), you can run failing tests with `DIALECT=mssql npm run test-docker-integration -- --grep test-title`